engines/wazero/wazero.go

```diff
@@ -60,201 +71,283 @@ type (
 	}
 )
 
-// Ensure the engine conforms to the waPC interface.
+// Ensure the runtime conforms to the waPC interface.
 var _ = (wapc.Module)((*Module)(nil))
 var _ = (wapc.Instance)((*Instance)(nil))
 
-var engineInstance = engine{}
+var runtimeInstance = runtime{}
 
 func Engine() wapc.Engine {
-	return &engineInstance
+	return &runtimeInstance
 }
 
-func (e *engine) Name() string {
+func (e *runtime) Name() string {
 	return "wazero"
 }
 
-// New compiles a `Module` from `code`.
-func (e *engine) New(code []byte, hostCallHandler wapc.HostCallHandler) (wapc.Module, error) {
-	store := wazero.NewStoreWithConfig(&wazero.StoreConfig{
-		Context: context.Background(),
-		Engine:  getEngine(),
-	})
-
-	module := &wazero.ModuleConfig{Source: code}
-	if err := module.Validate(); err != nil {
-		return nil, err
-	}
+type stdout struct {
+	// m acts as a field pointer to Module.wasiStdout until #19.
+	m *Module
+}
 
-	m := &Module{
-		store:           store,
-		module:          module,
-		hostCallHandler: hostCallHandler,
+// Write implements io.Writer by invoking the Module.writer or discarding if nil.
```

Comment:
hoping we can top-level io.Writer for stdout,stderr as it is more direct. Also clarify if this must be mutable